### PR TITLE
feat: imrpove refresh rate precision

### PR DESCRIPTION
**Pull request by Wanchai290**

### CODE REVIEW

**Review by benjaminchew4 — APPROVED** [review]
LGTM

**Comment by Wanchai290** [comment]
moving target branch to `main`, I don't think we'll use `develop` in its current state again, we'll just re-create it when we finish implementing the features.

**Comment by Wanchai290** [comment]
nevermind it's going to cause such a mess
maybe we should cherry-pick it and create a new PR

**Comment by Wanchai290** [comment]
Integrated in above PR by commit https://github.com/NAMeC-SSL/CRAbE/commit/1acd72f86332b555533b46531276ab219547cc5e  
Should we close this ?

File: crates/crabe/src/main.rs [hunk 1]
```diff
@@ -145,7 +147,14 @@ impl System {
             self.guard_component
                 .step(&mut self.world, &mut command_map, &mut ToolCommands);
             feedback = self.output_component.step(command_map, ToolCommands);
-            thread::sleep(_refresh_rate);
+            // info!("Execution time : {} μs", &timer.elapsed().as_micros());
+            let elapsed = timer.elapsed();
+            let mut sleep_time = _refresh_rate;
+            if elapsed <= _refresh_rate {
```
Comment (EtienneSchmitz) [inline crates/crabe/src/main.rs:153]: Il semble y avoir un problème si le temps `elasped` est supérieur à `refresh_rate`, la variable `sleep_time` sera égale au `refresh_rate`. 
Ce qui va donc impliquer un sleep (alors qu'il aurait du être égale à zéro).

**Review by EtienneSchmitz — CHANGES_REQUESTED** [review]

File: crates/crabe/src/main.rs (hunk 1 above)
Comment (EtienneSchmitz) [reply crates/crabe/src/main.rs:153]: Aussi vu qu'on utilse maintenant refresh_rate il faut supprimer le todo en haut du fichier.

Comment (EtienneSchmitz) [reply crates/crabe/src/main.rs:153]: Dans la précédente version de CRAbE, nous avions ça : 
```rust
let elapsed = start.elapsed().as_micros() as f64 / 1000.0;
let sleep_time = Duration::from_millis(16).as_micros() as i128 - start.elapsed().as_micros() as i128;
if sleep_time > 0 {
    sleep(Duration::from_micros(sleep_time as u64));
}
```

Comment (Wanchai290) [reply crates/crabe/src/main.rs:153]: Alright I'll let you commit that on this branch, and I'll cherry-pick it onto #75 if everyone agrees with this. I don't have time at the moment to test it out.

**Comment by Wanchai290** [comment]
moved to #94

**Pull request closed without merging**